### ISSUE TITLE
Yahoo Symbols Not Found

### ISSUE THREAD

**Issue by call-1511** [dpguthrie/yahooquery#139]
Wonderful work! Tried using YFinance, but yahooquery seems to work better.

I tried to obtain a list of tickers for companies around the world using the code snippet that you had written but for some companies it returned "Symbols not found" when they are in fact available:

```
import pandas as pd
import yahooquery as yq

def get_symbol(query, preferred_exchange=''):
    try:
        data = yq.search(query)
    except ValueError: # Will catch JSONDecodeError
        print(query)
    else:
        quotes = data['quotes']
        if len(quotes) == 0:
            return 'No Symbol Found'

        symbol = quotes[0]['symbol']
        for quote in quotes:
            if quote['exchange'] == preferred_exchange:
                symbol = quote['symbol']
                break
        return symbol

companies=['BP','BYD','Porsche','Airbus','Shenhua Energy','CNOOC','Nongfu','Yangtze Power','Foshan Haitian Flavoring','Zijin Mining','Wanhua Chemical','S.F. Express','Xiaomi','State Construction Engineering','Luxshare Precision','Great Wall Motors','Vanke','Yili Group','Yanzhuang Energy','Ganfeng Lithium','Anhui Coch','China Railway','Mengniu Dairy','China Southern Airlines','DiDi','CRRC','Aluminum Corp of China','China Oilfield Services','PetroChina','Adani Total Gas','Adani Enterprises','Larsen & Toubro','Asian Paints','UltraTech Cement','Oil & Natural Gas','JSW Steel','PowerGrid Corporation of India','Tata Steel','Coal India','Petrobas','Centrais Eletricas Brasileiras']

df = pd.DataFrame({'Company name': companies})
df['Company symbol'] = df.apply(lambda x: get_symbol(x['Company name']), axis=1)

```
**Expected behavior**
A full list of equity tickers should be retrieved, however, various companies do not have their tickers returned even though they can be retrieved through Yahoo Finance. For example:

- Great Wall Motors =  601633.SS
- Foshan Haitian Flavoring = 603288.SS
- PowerGrid Corporation of India = POWRGRIND.NS
- Petrobras = PBR

**Screenshots**
                      Company name   Company symbol
0                               BP               BP
1                              BYD              BYD
2                          Porsche            POAHY
3                           Airbus           AIR.PA
4                   Shenhua Energy          1088.HK
5                            CNOOC          0883.HK
6                           Nongfu          9633.HK
7                    Yangtze Power        600900.SS
8         Foshan Haitian Flavoring  No Symbol Found
9                     Zijin Mining          2899.HK
10                 Wanhua Chemical        600309.SS
11                    S.F. Express  No Symbol Found
12                          Xiaomi          1810.HK
13  State Construction Engineering        601668.SS
14              Luxshare Precision        002475.SZ
15               Great Wall Motors  No Symbol Found
16                           Vanke          2202.HK
17                      Yili Group  No Symbol Found
18                Yanzhuang Energy  No Symbol Found
19                 Ganfeng Lithium            GNENF
20                      Anhui Coch  No Symbol Found
21                   China Railway          0390.HK
22                   Mengniu Dairy          2319.HK
23         China Southern Airlines              ZNH
24                            DiDi            DIDIY
25                            CRRC          1766.HK
26          Aluminum Corp of China            ACHHY
27         China Oilfield Services          2883.HK
28                      PetroChina          0857.HK
29                 Adani Total Gas          ATGL.NS
30               Adani Enterprises      ADANIENT.NS
31                 Larsen & Toubro            LT.NS
32                    Asian Paints    ASIANPAINT.NS
33                UltraTech Cement    ULTRACEMCO.NS
34               Oil & Natural Gas          ONGC.NS
35                       JSW Steel      JSWSTEEL.NS
36  PowerGrid Corporation of India  No Symbol Found
37                      Tata Steel     TATASTEEL.NS
38                      Coal India     COALINDIA.NS
39                        Petrobas  No Symbol Found
40  Centrais Eletricas Brasileiras            EBR-B

**Comment by dpguthrie** [comment]
This is just a function of yahoo finance's search endpoint.  Put in "Great Wall Motors" into the search bar and you'll see that 0 quotes are returned.  Unfortunately, not much I'm able to do here as I'm just returning what the endpoint returns.  Gonna close out as this is expected behavior.

<img width="1370" alt="image" src="https://user-images.githubusercontent.com/10198967/215362459-4a3412f8-b9d2-4274-aded-fe4bba7fcd18.png">

Does look like this name should be "Great Wall Motor" and the appropriate quotes are returned.

**Comment by call-1511** [comment]
Thank you Doug for your great work. Much appreciated.
The errors arose due to typos on the company names. After carrying out the rectifications, all the symbols were returned!
Best Rgds,Desmond


    On Monday, January 30, 2023 at 07:37:07 AM GMT+8, Doug Guthrie ***@***.***> wrote:  
 
 


Closed #139 as completed.
